test/address-representation-converters.spec.ts

```diff
@@ -27,7 +33,44 @@ describe('simpleCellAddressFromString', () => {
   })
 
   it('should return address with sheet number from sheet mapping regardless of override parameter', () => {
-    expect(simpleCellAddressFromString(sheetMappingFunction, 'Sheet3!A1', 1)).toEqual(simpleCellAddress(2, 0, 0))
+    expect(simpleCellAddressFromString(sheetMappingFunction, 'Sheet3!A1', 1)).toEqual(adr('A1', 2))
+  })
+})
+
+describe('simpleCellRangeFromString', () => {
+  const sheetMappingFunction = (name: string): Maybe<number> => {
+    const index = ['Sheet1', 'Sheet2', 'Sheet3'].indexOf(name)
+    return index > 0 ? index : undefined
+  }
+
+  it('should not accept invalid ranges', () => {
+    expect(simpleCellRangeFromString(sheetMappingFunction, 'A1', 0)).toEqual(undefined)
+    expect(simpleCellRangeFromString(sheetMappingFunction, 'A1:B1:C1', 0)).toEqual(undefined)
+  })
+
+  it('should return undefined when the Sheet4 does not exist or does not match sheetId = 0?', () => {
```

Comment:
I've put here a question, it wasn't ready to merge as is

Reply:
oops :)